app/service/agent_svc.py

```diff
@@ -130,24 +129,30 @@ def __init__(self, untrusted_timer):
         :param link:
         :return: the id of the created link
         """
-        data_svc = self.get_service('data_svc')
         operation_svc = self.get_service('operation_svc')
         op_id = link['op_id']
 
-        operation = (await data_svc.dao.get('core_operation', dict(id=op_id)))[0]
+        operation = (await self.data_svc.dao.get('core_operation', dict(id=op_id)))[0]
         while operation['state'] != operation_svc.op_states['RUNNING']:
             if operation['state'] == operation_svc.op_states['RUN_ONE_LINK']:
-                link_id = await data_svc.create('core_chain', link)
-                await data_svc.dao.update('core_operation', 'id', op_id, dict(state=operation_svc.op_states['PAUSED']))
+                link_id = await self._create_link(link)
+                await self.data_svc.dao.update('core_operation', 'id', op_id, dict(state=operation_svc.op_states['PAUSED']))
                 return link_id
             else:
                 await asyncio.sleep(30)
-                operation = (await data_svc.dao.get('core_operation', dict(id=op_id)))[0]
+                operation = (await self.data_svc.dao.get('core_operation', dict(id=op_id)))[0]
         link.pop('adversary_map_id')
-        return await data_svc.create('core_chain', link)
+        return await self._create_link(link)
 
     """ PRIVATE """
 
+    async def _create_link(self, link):
```

Comment:
this function seems like it would be better suited on the data_svc as a public function. thoughts?